### PR TITLE
Sponsor/Partner updates

### PR DESCRIPTION
**Pull request by ellisonbg**
Updates the following information:

* Changes my affiliation to AWS
* Adds AWS as an institutional partner (by virtue of my being there now). Cal Poly remains as a partner due to Ana.
* Adds Schmidt Futures and Quansight as sponsors. Cal Poly and NYU have a grant from Schmidt to fund work on "rich context" for JupyterLab (commenting, data registry, metadata work). Quansight has been making in kind donations of parts of some core contributor's time (Saul, Ian, Grant, maybe others).

### CODE REVIEW

**Comment by Ruv7** [comment]
Brian - can you help me out by reminding me how to preview these changes?

**Comment by ellisonbg** [comment]
Click on "Show all checks" and then click on the "Details" of `deploy/netlify`. However, there is a bug in the asset path in netlify that makes it so the new logos won't show up on that build. I will post a screenshot from my local build once I get that working.

**Comment by ellisonbg** [comment]
Got it building locally, fixing some bugs.

**Comment by Carreau** [comment]
> However, there is a bug in the asset path in netlify that makes it so the new logos won't show up on that build

`site.baseurl` instead of `site.url`. I've fixed it for the two logos that were added.

**Comment by Carreau** [comment]
Also schmidt is png not svg. Fixed as well.

**Comment by ellisonbg** [comment]
@Carreau thanks! I spent the afternoon wrestling with the bootstrap grid system. Somehow, adding just one more partner logo caused our already broken usage of the grid system to actually break. Was painful, but I think we are at least semi-using it correctly now. Here are screenshots:

Screen Shot 2019-03-20 at 3 15 44 PM

Further down:

Screen Shot 2019-03-20 at 3 15 58 PM

Further down:

Screen Shot 2019-03-20 at 3 16 11 PM

**Comment by ellisonbg** [comment]
...had to rebase on your changes so had to force push...

**Comment by ellisonbg** [comment]
OK everything is rendering fine now.

File: _includes/partners.html [hunk 1]
```diff
@@ -1,57 +1,66 @@
...
-            </div>
-            <div class="col-sm-4 col-xs-6 sponsor ">
-                <a href="https://www.jpmorganchase.com/">
-                    <img src="{{site.url}}/assets/jpmc_logo.png" class="company-logo" alt="JPMorgan Chase logo">
-                </a>
-            </div>
-            <div class="col-sm-4 col-xs-6 sponsor">
+        </div>
+    </div>
+    <div class="row">
```
Comment (Carreau) [inline _includes/partners.html]: Wraping in `row`s won't work on XS, in here you are assuming there are 4 logo per line; though on narrow screen there are only two. 
Screen Shot 2019-03-20 at 3 27 15 PM

**Comment by Carreau** [comment]
This could also likely be easier to manipulate if it was templated as well. I'm also pretty sure the solution is not rows, but to force the min-height/width of each log. 

Here is a row-less with fixed width/height crudely done in the inspector: 

--- 
Screen Shot 2019-03-20 at 3 44 20 PM
--- 

Screen Shot 2019-03-20 at 3 44 12 PM

--- 

Screen Shot 2019-03-20 at 3 44 05 PM

Happy to merge what you did, but I believe that will just be more technical debt and edge cases.

**Comment by Carreau** [comment]
@ellisonbg I have attempted to show you what I mean in #329. Seem to behave a bit better to me; and feel simpler.

**Comment by ellisonbg** [comment]
I added a comment on your PR, but will give more details here. The problem is that the current DOM structure doesn't follow the bootstrap grid model in terms of which classes are used where. The basic idea is that columns should be inside rows, which would be inside container. The old version had these classes all mixed together in inconsistent ways. The result is that the layout was failing in odd ways and was very fragile. I agree that we should move towards separating out the data/model into yml data. But that is a separate matter from the work done here.

Can you clarify how this adds more technical debt and edge cases?

**Comment by Carreau** [comment]
> The basic idea is that columns should be inside row**S**, which would be inside container.
> Can you clarify how this adds more technical debt and edge cases?

I do not disagree on this per say but with your interpretation.
Note the emphasis of mine on the rows plural.

It was my understanding that using **multiple** rows is useful only if you know **at all time** how many items per rows you have and if **this number does not change**. You also over-constrain in your understanding and in your implementation with "the sum of columns widths in each row should be 12". Which is definitely not true according to boostrap docs

Bootstrap 3.3 introduction: 
  > If more than 12 columns are placed within a single row, each group of extra columns will, as one unit, wrap onto a new line.

So when you have _more than 12 elements_ depending on your case you might want a single "row" div. I agree that the getbootstrap.com does not show a single example of this, and do always have `sum(width)<=12` which I understand can add to your confusion.

In our case we are switching from 2 columns/"row" on xs to 4 column per "row" on md. In this case You should avoid baking into the HTML multiple "row" divs to avoid reflow issues. In our case it is hidden as 2 is a divider of 4. but on a md-5/xs-3 for example this would lead to the following with rows on md: 

```
xxxxx
xxxxx
```

And this on xs:
```
xxx
xx
xxx
xx
```

While it would reflow properly w/o rows.  If we had only one of `col-*-3` or `col-x*-6` I would agree with you; But as we have both classes I think we are just pushing what we ask of bootstrap into what it is not meant to do. 

And the issues are shown on the bootstrap website:  

Screen Shot 2019-03-21 at 7 55 11 AM

Here is an  the same, but with all columns in a single "row" div modified in inspector: 

Screen Shot 2019-03-21 at 8 09 45 AM



Funnily enough what we are trying to achieve is done on the bootstrap website [w/o grid](https://getbootstrap.com/docs/3.3/components/): 


Screen Shot 2019-03-21 at 7 58 01 AM


And our fix for items of different hight is not to wrap in multiple rows, but to use column reset. 

Screen Shot 2019-03-21 at 7 56 13 AM


So to summarize, I think you are over-interpreting the "columns" in "rows", and that your implementation looking almost quite right is a fluke. Hope that clarify my thinking and understanding.

**Comment by ellisonbg** [comment]
The issue I was seeing initially is that the "auto wrapping" was making some really awkward decisions, essentially trying to wedge two logos in to 1 space in the grid (should have taken a screenshot). I started to read the bootstrap docs and could not understand how our usage was consistent with their spec.

In particular, we were not following these rules:

> Rows must be placed within a .container (fixed-width) or .container-fluid (full-width) for proper alignment and padding.
> Use rows to create horizontal groups of columns.
> Content should be placed within columns, and only columns may be immediate children of rows.

We had non-columns children of rows, rows weren't children of containers, nested columns within columns (which is allowed, but we weren't using it as intended), and multiple levels between the container and the rows. Put another way, I read the spec and being designed for things like this:

```
<div class="container">
  <div class="row">
    <div class="col-xs-4">...</div>
    <div class="col-xs-4">...</div>
    ...
  </div>
  ....
</div>
```

Whereas we had something more like:

```
<div class="not a container">
  <div class="row">
    ...children that are not columns...
    <div class="col-xs-12 section-center">
        <div class="col-sm-4 col-xs-6 sponsor">
        ...
    </div>
  </div>
  ...
</div>
```

In trying to debug the layout problems, it was difficult to follow/debug/understand the boostrap spec, when were weren't following it. Along the way I tried to put fixed numbers of things into rows, and clean up the DOM structure, and it eventually started to work in a manner I could reason about.

What I haven't tried is to go back to a single auto-wrapping row with this newly refactored overall DOM structure. More than willing to give that a go. I should also note that there are still minor ways we aren't following the spec (rows are not direct children of containers), but fixing that would require even more substantial changes. I will try to simplify it a bit though and see how it goes.

**Comment by ellisonbg** [comment]
OK here is an example of what I was referring to. I have arranged the sponsors into a single row with `md` and `xs` classes for desktop and mobile. Here is the collapsed DOM structure:

Screen Shot 2019-03-22 at 7 43 55 PM

And the result:

Screen Shot 2019-03-22 at 7 43 44 PM

I am sure if I dug into the entire bootstrap model and all our css I could figure out what is going on, but it is hard to want to invest the effort to become an expert in the bootstrap 3.x grid model

Open to further suggestions :-)

**Comment by Carreau** [comment]
> ...
Fair enough, this is at least moving things forward, and if it appear to not work with a ingle row, I suggest we iterate on the CSS in subsequent PR. 

Thanks !
